back/server.js

```diff
@@ -15,250 +15,61 @@ app.use(
-		title: 'Je quitte mon appart',
-		category: 'Logement',
-		content:
-			"Hey les copains, je quitte mon appartement car je retourne chez mes parents, c'est un F2 en plein centre ville (plus d'info sur www.logement.com).",
-		city: 'Lille',
-	},
-	{
-		id: 7,
-		user_id: 7,
-		firstname: 'Youri',
-		lastname: 'Ligotmi',
-		studies: 'Faculté de psychologie',
-		picture:
-			'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTJ1cCNodtCmfkN2KKqQx84RveyD2FYR8QS--HYkopSs3Uh3x36&s',
-		title: 'Cours de Tai-chi cosmique',
-		category: 'Cours',
-		content:
-			'Ami du grand monarque bonjour, je donne des cours particulier de tai-chi cosmique, ce qui vous permettra de réaliser une experience astral hors du commun. Si vous etes interesse contactez moi au plus vite (place limitée).',
-		city: 'Reims',
-	},
-];
-
 app.get('/posts', (req, res) => {
-	res.send(posts);
+	let sqlQuery = 'SELECT user.firstname, user.lastname, user.city, post.* FROM post JOIN user ON user.id=post.user_id'
+  	connection.query(sqlQuery, (err, results) => {
+    if (err) {
+      res.status(500).send('Erreur lors de la récupération des posts');
+    } else {
+      res.json(results);
+    }
+  });
 });
 
 app.post('/posts', (req, res) => {
-	const formData = req.body;
-	const post = {
-		id: formData.id,
-		user_id: formData.user_id,
-		firstname: formData.firstname,
-		lastname: formData.lastname,
-		studies: formData.studies,
-		picture: formData.picture,
-		title: formData.title,
-		category: formData.category,
-		content: formData.content,
-		city: formData.city,
-	};
-	posts.push(post);
-	res.send(post);
-	if (err) {
-		console.log(err);
-		res.status(500).send('Error posting');
-	} else {
-		res.sendStatus(200);
-	}
+  	const formData = req.body;
+	let sqlQuery = 'INSERT INTO post SET ?';
+  	connection.query(sqlQuery, formData, (err, results) => {
+    if (err) {
+      console.log(err);
+      res.status(500).send("Error sending a new post");
+    } else {
+      res.sendStatus(200);
+    }
+  	});
 });
 
 app.post('/login', (req, res) => {
-	const formData = req.body;
-	const userEmail = formData.email;
-	const userTel = formData.tel;
-	const userPassword = formData.password;
-	let match = 0
-	for (let i = 0 ; i < users.length ; i++) {
-		if (userEmail === users[i].email) {
-			if (users[i].password === userPassword) {
-				res.status(200).send('Login succeeded')
+	let userPassword = req.body.password;
+	let userEmail = req.body.email
+	let sqlQuery = `SELECT email, password from user where user.email='${userEmail}'`
+	connection.query(sqlQuery, (err, results) => {
+		if (!results.length) {
+			res.status(400).send('Wrong email or password')
+		} else {
+			if (results[0].password === userPassword) {
+				res.sendStatus(200)
 			} else {
-				res.send('Wrong username or password')
+				res.status(400).send('Wrong email or password')
 			}
-			match++
 		}
-	}
-	if (match === 0) {
-		res.send('Wrong username or password')
-	}
-});
+		if (err) {
```

Comment:
Can you ever get to this line of code ?

Reply:
I assumed we could get to this line of code in case something goes wrong when accessing the database... Should I remove it ?